tcmalloc/internal/percpu_rseq_aarch64.S

```diff
@@ -474,9 +496,28 @@ TcmallocSlab_Pop:
     ret
 .LTcmallocSlab_Pop_no_items:
     mov x0, x8                  /* call overflow handler with CPU ID */
-    br x2
+    TAILCALL(x2)
   .cfi_endproc
 ENCODE_SIZE(TcmallocSlab_Pop)
 DEFINE_UPSTREAM_CS(TcmallocSlab_Pop)
 
 .section .note.GNU-stack,"",@progbits
+
+/* Add a NT_GNU_PROPERTY_TYPE_0 note.  */
+#define GNU_PROPERTY(type, value)       \
+  .section .note.gnu.property, "a";     \
+  .p2align 3;                           \
+  .word 4;                              \
+  .word 16;                             \
+  .word 5;                              \
+  .asciz "GNU";                         \
+  .word type;                           \
+  .word 4;                              \
+  .word value;                          \
+  .word 0;
+
+/* Add GNU property note if built with branch protection.  */
+
+#if defined(__ARM_FEATURE_BTI_DEFAULT)
+GNU_PROPERTY (0xc0000000, 1)
```

Comment:
I haven't seen the use of gnu properties before, so I had a question about this: why did you choose these values for the type and value?

Reply:
I hadn't seen it before either, but IIUC this is added such that the object file resulting from the assembly of this file contains a .note.gnu.property NOTE section, and these values indicate the code is BTI enabled. The linker, by default, checks whether all included object files contain such a note. If that is the case then the resulting binary will also be marked as BTI-enabled, which I believe means that the loader will then set the appropriate tag to indicate it is a 'guarded memory region'.

This is why I used '-z force-bti' as then the linker will insert the note section in the final binary regardless and warn us about what object files were missing it. In my case those were all boot-up code that had not been built with BTI enabled. For C/C++ code the compiler will create the note for you if you tell it to enable BTI.

As to answer the original question as to why these values, I looked at other uses of this note section in assembly to get the approrpriate values, for instance here:
https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=libgcc/config/aarch64/lse.S;h=7496b8b6337fb2c924ba338046df396522aa0a85;hb=refs/heads/master#l281

Reply:
Thanks for that very helpful explanation!

Reply:
I was puzzled by the other note, but it seems like both are needed ([based on this similar LLVM commit](https://reviews.llvm.org/rGd53e61863d48a07ce285d5b0a36abc67583023bd))